sambacc/commands/ctdb.py

```diff
@@ -271,17 +271,19 @@ def _lookup_hostname(hostname: str) -> str:
         )
         ipv6_address = None
 
-        for entry in addrinfo:
-            family, _, _, _, sockaddr = entry
-            ip_address = sockaddr[0]
-
-            if ip_address.startswith("127.") or ip_address == "::1":
-                continue
-
+        for family, _, _, _, sockaddr in addrinfo:
             if family == socket.AF_INET:
+                ip_address = sockaddr[0]
+                assert isinstance(ip_address, str)
+                if ip_address.startswith("127."):
+                    continue
                 return ip_address
 
             if family == socket.AF_INET6 and ipv6_address is None:
+                ip_address = sockaddr[0]
+                assert isinstance(ip_address, str)
+                if ip_address == "::1":
+                    continue
                 ipv6_address = ip_address
```

Comment:
@avanthakkar Why are we not returning IPv6 address from here? What special case are we dealing with additional `ipv6_address`?

Reply:
> @avanthakkar Why are we not returning IPv6 address from here? What special case are we dealing with additional `ipv6_address`?

Yeah you're right, we probably don't need to explicitly check for types, probably this is enough?
```
..............
ip_address = sockaddr[0]
assert isinstance(ip_address, str)

if ip_address.startswith("127.") or ip_address == "::1":
    continue
            
return ip_address
..................
```

Reply:
> ```
> ..............
> ip_address = sockaddr[0]
> assert isinstance(ip_address, str)
> 
> if ip_address.startswith("127.") or ip_address == "::1":
>     continue
>             
> return ip_address
> ..................
> ```

I've created #147 in the above manner but retaining the check for address info family type.